### PR TITLE
desktop: Move `cpal` audio backend to `frontend-utils`

### PR DESCRIPTION
**Pull request by madsmtm**
Required for https://github.com/madsmtm/ruffle-ios.

### CODE REVIEW

**Review by Dinnerbone — APPROVED** [review]
Thank you!

File: frontend-utils/src/backends/audio.rs [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+    DefaultStream(#[from] cpal::DefaultStreamConfigError),
+
+    #[error("Unsupported sample format {0:?}")]
+    UnsupportedSampleFormat(SampleFormat),
+
+    #[error("Couldn't play the audio stream")]
+    Play(#[from] cpal::PlayStreamError),
+
+    #[error("failed constructing audio stream")]
+    Build(#[from] cpal::BuildStreamError),
```
Comment (torokati44) [inline frontend-utils/src/backends/audio.rs:24]: nit: Capitalization - but also, is this actually used?

Comment (torokati44) [reply frontend-utils/src/backends/audio.rs:24]: it is, implicitly.

Comment (torokati44) [reply frontend-utils/src/backends/audio.rs]: Fixed the capitalization, and changed up the phrasing a little bit - I hope you don't mind!

**Review by torokati44 — APPROVED** [review]

**Pull request merged**